src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -272,6 +272,14 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node,
                 return true;
             }
 
+            // Range expression
+            // x[(y)..] => x[y..]
+            if (node.Parent is RangeExpressionSyntax)
+            {
+                if (expression.IsKind(SyntaxKind.ElementAccessExpression))
+                    return true;
+            }
```

Comment:
❔ Why is this getting added? It seems like the call to `RemovalChangesAssociation` would already be responsible for handling this case.

Reply:
I was expecting a comment on this one :)

The current code doesn't remove the parentheses in the following case `s[(a[0])..]`. `RemovalChangesAssociation` is conservative as `RangeExpressionSyntax` precedence is `None`. I think I need to add `OperatorPrecedence.Range` between `Multiplicative` and `Unary` ([doc](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/#operator-precedence)) to fix the issue.

In `GetOperatorPrecedence` there are comments that seems to come from the C# spec (`// From C# spec, 7.3.1:`). Where is the URL of this spec so I can add a similar comment for range?

Reply:
>  I think I need to add OperatorPrecedence.Range between Multiplicative and Unary (doc) to fix the issue.

Yes, that would make sense to me.  ranges are here: https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/ranges.md